src/mcfret.c

```diff
@@ -0,0 +1,366 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <string.h>
+#include <time.h>
+#include <fftw3.h>
+#include "omp.h"
+#include "types.h"
+#include "NISE_subs.h"
+#include "mcfret.h"
+#include "project.h"
+
+
+
+
+   
+void mcfret(t_non *non){
+  /* Call the MCFRET Routine */
+   if (!strcmp(non->technique, "MCFRET") || (!strcmp(non->technique, "MCFRET-Autodetect")) || (!strcmp(non->technique, "MCFRET-Absorption"))
+        || (!strcmp(non->technique, "MCFRET-Emission")) || (!strcmp(non->technique, "MCFRET-Coupling")) || (!strcmp(non->technique,
+        "MCFRET-Rate")) || (!strcmp(non->technique, "MCFRET-Analyse")) ) {
+        printf("Perfroming the MCFRET calculation.\n");
+    }
+
+/* Define variables and arrays */
+   /* Integers */
+    int nn2;
+    int itime,N_samples;
+    int samples;
+    int x,ti,tj,i,j;
+    int t1,fft;
+    int elements;
+    int cl,Ncl;
+
+    /* Response functions for emission and absorption: real and imaginary part*/
+    float *re_S_Abs,*im_S_Abs;
+    float *re_S_Emi,*im_S_Emi;
+    /*Hamiltonian of the whole system - all donors and acceptors included*/
+    float *Hamil_i_e;
+    /*Vectors representing time dependent states: real and imaginary part*/
+    float *vecr, *veci;
+    /*Density matrix*/
+    float *dens_mat;
+
+    float shift1;
+  /* 1D Fourier transform */
+    fftw_complex *fftIn,*fftOut;
+    /*was is das?*/
+    fftw_plan fftPlan;  
+
+  /* Time parameters */
+    time_t time_now,time_old,time_0;
+  /* Initialize time */
+    time(&time_now);
+    time(&time_0);
+    shift1=(non->max1+non->min1)/2;
+    printf("Frequency shift %f.\n",shift1);
+    non->shifte=shift1;
+
+
+  /*File handles*/
+
+    FILE *H_traj,*mu_traj;
+    FILE *C_traj;
+    FILE *absorption_matrix, *emission_matrix,*log;
+    FILE *Cfile;
+
+    /*Allocate memory for all the variables*/
+    /*Allocate memory for the response functions*/
+    nn2=non->singles*(non->singles+1)/2;
+    re_S_Abs=(float *)calloc(nn2*non->tmax,sizeof(float));
+    im_S_Abs=(float *)calloc(nn2*non->tmax,sizeof(float));
+    re_S_Emi=(float *)calloc(nn2*non->tmax,sizeof(float));
+    im_S_Emi=(float *)calloc(nn2*non->tmax,sizeof(float));
+
+    /*Allocating memory for the real and imaginary part of the wave function that we need to propagate*/
+    vecr=(float *)calloc(non->singles*non->singles,sizeof(float));	
+    veci=(float *)calloc(non->singles*non->singles,sizeof(float));
+
+    /*Allocating memory for the density matrix*/
+    dens_mat=(float *)calloc(non->singles*non->singles,sizeof(float));
+
+ /* Open Trajectory files */
+    H_traj=fopen(non->energyFName,"rb");
+    if (H_traj==NULL){
+        printf("Hamiltonian file not found!\n");
+        exit(1);
+    }
+      /*Here we want to call the routine for checking the trajectory files*/ 
```

Comment:
This comment is outdated. We actually do call the control routine. A comment explaining what it checks might be better.